src/webpub_manifest_parser/core/ast.py

```diff
@@ -0,0 +1,831 @@
+        illustrator=None,
+        letterer=None,
+        penciler=None,
+        colorist=None,
+        inker=None,
+        narrator=None,
+        contributor=None,
+        publisher=None,
+        imprint=None,
+        subject=None,
+        description=None,
+        duration=None,
+        number_of_pages=None,
+        belongs_to=None,
+    ):
+        """Initialize a new instance of Metadata class."""
+        super(Metadata, self).__init__()
+        self.title = title
+        self.identifier = identifier
+        self.title = title
+        self.subtitle = subtitle
+        self.modified = modified
+        self.published = published
+        self.language = language
+        self.sort_as = sort_as
+        self.author = author
+        self.translator = translator
+        self.editor = editor
+        self.artist = artist
+        self.illustrator = illustrator
+        self.letterer = letterer
+        self.penciler = penciler
+        self.colorist = colorist
+        self.inker = inker
+        self.narrator = narrator
+        self.contributor = contributor
+        self.publisher = publisher
+        self.imprint = imprint
+        self.subject = subject
+        self.description = description
+        self.duration = duration
+        self.number_of_pages = number_of_pages
+        self.belongs_to = belongs_to
+
+    def __hash__(self):
+        """Calculate the hash.
+
+        :return: Hash
+        :rtype: int
+        """
+        return hash(
+            (
+                self.identifier,
+                self.type,
+                self.title,
+                self.subtitle,
+                self.modified,
+                self.published,
+                self.language,
+                self.sort_as,
+                self.author,
+                self.translator,
+                self.editor,
+                self.artist,
+                self.illustrator,
+                self.letterer,
+                self.penciler,
+                self.colorist,
+                self.inker,
+                self.narrator,
+                self.contributor,
+                self.publisher,
+                self.imprint,
+                self.subject,
+                self.description,
+                self.duration,
+                self.number_of_pages,
+                self.belongs_to,
+            )
+        )
+
+
+class PresentationMetadata(Metadata):
+    """RWPM extension containing presentation metadata."""
+
+    clipped = BooleanProperty("clipped", False)
+    continuous = BooleanProperty("continuous", False)
+    fit = EnumProperty("fit", False, ["width", "height", "contain", "cover"])
+    orientation = EnumProperty("orientation", False, ["auto", "landscape", "portrait"])
+    overflow = EnumProperty(
+        "overflow", False, ["auto", "paginated", "scrolled", "scrolled-continuous"]
+    )
+    spread = EnumProperty("spread", False, ["auto", "both", "none", "landscape"])
+
+
+class CompactCollection(Node):
```

Comment:
On mobile we didn't make a difference between compact or full RWPM collections, everything is made canonical in a single full `Collection` type, with empty metadata. Now I'm not sure this is necessarily better, I just wanted to point out the difference.

Also the role is not part of the collection itself but of its parent, which contains a `subcollections` property which is a `Map` of `List<Collection>`: 
* https://github.com/readium/r2-shared-kotlin/blob/alpha/r2-shared/src/main/java/org/readium/r2/shared/publication/PublicationCollection.kt#L33
* https://github.com/readium/r2-shared-kotlin/blob/34c27504e93714d4c3f6ecfbc60b71913a01d659/r2-shared/src/main/java/org/readium/r2/shared/publication/Publication.kt#L86

Reply:
Leonard also used the same approach with having a single `Collection` class. In my approach we can rely on type but in a dynamic language like Python it doesn't make much sense